src/averageAges.js

```diff
@@ -55,7 +71,33 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people.filter(person => isHasChildren(people, person.name));
+  const children = people
+    .filter(child => {
+      const motherNames = mothers.map(mom => mom.name);
+
+      return motherNames
+        .includes(child.mother) && (!onlyWithSon || child.sex === 'm');
+    });
+  const differenceAges = children.map(child => {
+    const mother = mothers.find(mom => mom.name === child.mother);
+
+    return child.born - mother.born;
+  });
+  const totalDifferenceAges = differenceAges.reduce((a, b) => a + b, 0);
+
+  return (totalDifferenceAges / differenceAges.length) || 0;
```

Comment:
You use `reduce` and division by length in every function. Maybe we should move this logic to a separate helper?